File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxima(TreeNode* root,int maxval){
        if(!root) return 0;
        int res=(root->val>=maxval)?1:0;
        maxval=max(maxval,root->val);
        res+=maxima(root->left,maxval);
        res+=maxima(root->right,maxval);
        return res;
    }
    int goodNodes(TreeNode* root) {
        return maxima(root,root->val);
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        op1=self.diameterOfBinaryTree(root.left)
        op2=self.diameterOfBinaryTree(root.right)
        op3=self.height(root.left)+self.height(root.right)
        res=max(op1,max(op2,op3))
        return res
    def height(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        h1=self.height(root.left)
        h2=self.height(root.right)
        ans=max(h1,h2)+1
        return ans